otaclient/ota_proxy/__init__.py

```diff
@@ -13,9 +13,127 @@
 # limitations under the License.
 
 
+import asyncio
+import logging
+import multiprocessing
+from functools import partial
+from multiprocessing.context import SpawnProcess
+from pathlib import Path
+from typing import Callable
+
 from .cache_control import OTAFileCacheControl
 from .server_app import App
 from .ota_cache import OTACache, OTACacheScrubHelper
 from .config import config
 
-__all__ = ("App", "OTACache", "OTACacheScrubHelper", "OTAFileCacheControl", "config")
+logger = logging.getLogger(__name__)
+
+
+__all__ = (
+    "App",
+    "OTACache",
+    "OTACacheScrubHelper",
+    "OTAFileCacheControl",
+    "config",
+    "subprocess_start_otaproxy",
+)
+
+
+def _subprocess_main(
+    host: str,
+    port: int,
+    *,
+    init_cache: bool,
+    cache_dir: str,
+    cache_db_f: str,
+    upper_proxy: str,
+    enable_cache: bool,
+    enable_https: bool,
+    subprocess_init: Callable,
+):
+    import uvloop
+
+    # ------ pre-start callable ------ #
+    subprocess_init()
+    # ------ scrub cache folder if cache re-use is possible ------ #
+    should_init_cache = (
+        init_cache or not Path(cache_dir).is_dir() or not Path(cache_db_f).is_file()
+    )
+
+    if not should_init_cache:
+        scrub_helper = OTACacheScrubHelper(cache_db_f, cache_dir)
+        try:
+            scrub_helper.scrub_cache()
+        except Exception as e:
+            logger.error(f"scrub cache failed, force init: {e!r}")
+            should_init_cache = True
+        finally:
+            del scrub_helper
+
+    uvloop.install()
+    asyncio.run(
+        run_otaproxy(
+            host=host,
+            port=port,
+            cache_dir=cache_dir,
+            cache_db_f=cache_db_f,
+            enable_cache=enable_cache,
+            upper_proxy=upper_proxy,
+            enable_https=enable_https,
+            init_cache=should_init_cache,
+        )
+    )
+
+
+def subprocess_start_otaproxy(*args, **kwargs) -> SpawnProcess:
+    """Helper method to launch otaproxy in subprocess.
+
+    This method works like a wrapper and passthrough all args and kwargs
+    to the _subprocess_main function, and then execute the function in
+    a subprocess.
+    check _subprocess_main function for more details.
+    """
+
+    # run otaproxy in async loop in new subprocess
+    mp_ctx = multiprocessing.get_context("spawn")
+    otaproxy_subprocess = mp_ctx.Process(
+        target=partial(_subprocess_main, *args, **kwargs),
+        daemon=True,  # kill otaproxy if otaclient exists
```

Comment:
This function doesn't care about parent process is otaclient or not.
```suggestion
        daemon=True,  # kill otaproxy if the parent process exists
```